unitycatalog-ai/integrations/crewai/src/ucai_crewai/toolkit.py

```diff
@@ -0,0 +1,194 @@
+    get_tool_name,
+    process_function_names,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class UnityCatalogTool(CrewAIBaseTool):
+    """
+    A tool class that integrates Unity Catalog functions into a tool structure.
+
+    Attributes:
+        fn (Callable): A function to override the _run method of CrewAI.
+        client_config (Dict[str, Any]): Configuration settings for managing the client.
+    """
+
+    fn: Callable = Field(description="Callable that will override the CrewAI _run() method.")
+    client_config: Dict[str, Any] = Field(
+        description="Configuration of the client for managing the tool"
+    )
+
+    def __init__(self, fn: Callable, client_config: Dict[str, Any], **kwargs):
+        """
+        A tool class that integrates Unity Catalog functions into a tool structure.
+
+        Args:
+            fn (Callable): The function that represents the tool's functionality.
+            metadata (ToolMetadata): Metadata about the tool, including name, description, and schema.
+            client_config (Dict[str, Any]): Configuration dictionary for the client used to manage the tool.
+            *args: Additional positional arguments.
+            **kwargs: Additional keyword arguments.
+        """
+        super().__init__(fn=fn, client_config=client_config, **kwargs)
+
+    def _run(self, *args: Any, **kwargs: Any) -> Any:
+        """Override of the CrewAI BaseTool run method."""
+        return self.fn(*args, **kwargs)
+
+
+class UCFunctionToolkit(BaseModel):
+    """
+    A toolkit for managing Unity Catalog functions and converting them into tools.
+
+    Attributes:
+        function_names (List[str]): List of function names in 'catalog.schema.function' format.
+        tools_dict (Dict[str, FunctionTool]): A dictionary mapping function names to their
+            corresponding tools.
+        client (Optional[BaseFunctionClient]): The client used to manage functions.
+        description_updated (Optional[Bool]): Flag to check if the description has been updated.
+        cache_function (Optional[Callable]): Function that will be used to determine if the tool
+            should be cached, should return a boolean. If None, the tool will be cached.
+        result_as_answer (Optional[Bool]): Flag to check if the tool should be the final
+            agent answer.
+
+    """
+
+    function_names: List[str] = Field(
+        default_factory=list,
+        description="List of function names in 'catalog.schema.function' format",
+    )
+    tools_dict: Dict[str, CrewAIBaseTool] = Field(default_factory=dict)
+    client: Optional[BaseFunctionClient] = Field(
+        default=None,
+        description="Client for managing functions",
+    )
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    # CrewAI parameters, which can be found in the link below
+    # https://github.com/crewAIInc/crewAI-tools/blob/main/crewai_tools/tools/base_tool.py#L21
+    description_updated: Optional[bool] = Field(
```

Comment:
```suggestion
    description_updated: bool = Field(
```
as the default value is False

Reply:
If default value is `False`, the param is optional, no?